score/etl/score/etl.py

```diff
@@ -0,0 +1,378 @@
+import collections
+import functools
+import pandas as pd
+
+from etl.base import ExtractTransformLoad
+from utils import get_module_logger
+from etl.sources.census.etl_utils import get_state_fips_codes
+
+logger = get_module_logger(__name__)
+
+
+class ScoreETL(ExtractTransformLoad):
+    def __init__(self):
+        # Define some global parameters
+        self.GEOID_FIELD_NAME = "GEOID10"
+        self.GEOID_TRACT_FIELD_NAME = "GEOID10_TRACT"
+        self.BUCKET_SOCIOECONOMIC = "Socioeconomic Factors"
+        self.BUCKET_SENSITIVE = "Sensitive populations"
+        self.BUCKET_ENVIRONMENTAL = "Environmental effects"
+        self.BUCKET_EXPOSURES = "Exposures"
+        self.BUCKETS = [
+            self.BUCKET_SOCIOECONOMIC,
+            self.BUCKET_SENSITIVE,
+            self.BUCKET_ENVIRONMENTAL,
+            self.BUCKET_EXPOSURES,
+        ]
+
+        # A few specific field names
+        # TODO: clean this up, I name some fields but not others.
+        self.UNEMPLOYED_FIELD_NAME = "Unemployed civilians (percent)"
+        self.LINGUISTIC_ISOLATION_FIELD_NAME = "Linguistic isolation (percent)"
+        self.HOUSING_BURDEN_FIELD_NAME = "Housing burden (percent)"
+        self.POVERTY_FIELD_NAME = "Poverty (Less than 200% of federal poverty line)"
+        self.HIGH_SCHOOL_FIELD_NAME = (
+            "Percent individuals age 25 or over with less than high school degree"
+        )
+
+        # There's another aggregation level (a second level of "buckets").
+        self.AGGREGATION_POLLUTION = "Pollution Burden"
+        self.AGGREGATION_POPULATION = "Population Characteristics"
+
+        self.PERCENTILE_FIELD_SUFFIX = " (percentile)"
+        self.MIN_MAX_FIELD_SUFFIX = " (min-max normalized)"
+
+        self.SCORE_CSV_PATH = self.DATA_PATH / "score" / "csv"
+
+        # dataframes
+        self.df: pd.DataFrame
+        self.ejscreen_df: pd.DataFrame
+        self.census_df: pd.DataFrame
+        self.housing_and_transportation_df: pd.DataFrame
+        self.hud_housing_df: pd.DataFrame
+
+    def extract(self) -> None:
+        # EJSCreen csv Load
+        ejscreen_csv = self.DATA_PATH / "dataset" / "ejscreen_2020" / "usa.csv"
```

Comment:
@esfoobar-usds -- this comment is what I'm referring to about `get_df`. 

Instead of requiring the score code to know the location of the `ejscreen_csv` and load it, what if the `ExtractTransformLoad` class had a method `get_df`? Then for `EJScreenETL`, `EJScreenETL.get_df` would either load the file from disk or, if it doesn't exist yet, run the ETL and then load the file from disk. 

This would reduce the code needed in the scoring function, as well as reducing the need to keep the CSV file paths in sync in multiple places.  

Then we could rewrite as follows: 

```
def extract(self) -> None:
    self.ejscreen_df = EJScreenETL.get_df()
    self.census_df = CensusETL.get_df()
    ...
```

Does that make sense?

Reply:
Got you! Yeah, I thought about it and this would infer instantiating ETL classes from the Score ETL, which is in itself an ETL process, so I'd rather call this from an outside controller, which in our case it's `runner.py`. We might run different data set ETLs in tandem or potentially not have the score and the ETL processes running on the same box, so unless I understand a more clear advantage of doing this, I would leave this one out of this PR.

An idea that did spark from this comment is to get the CSV location from a YAML file property, so further discussion once we get that piece going.